### PR TITLE
Pass in ajv instance

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,19 +4,22 @@ const fp = require('fastify-plugin')
 const Ajv = require('ajv')
 
 function fastifyResponseValidation (fastify, opts, next) {
+  const isAjvInstance = opts.ajv && opts.ajv instanceof Ajv
```
Comment (Eomm) [inline index.js:7]: If the user provides its own ajv instance, I would skip all the code below and we need to set the `ajv` var only (no plugins, nor `new Ajv`)

Comment (dancastillo) [reply index.js]: Sounds good. Updated code should reflect this now

File: test/ajv.test.js [hunk 2]
```diff
@@ -74,3 +77,78 @@ test('should throw an error if ajv.plugins is not an array', async t => {
...
+    }
+  })
+
+  const response = await fastify.inject({
+    method: 'GET',
+    url: '/'
+  })
+
+  t.equal(response.statusCode, 500)
+  t.equal(JSON.parse(response.payload).message, 'response should be an object with an integer property answer only')
```
Comment (Eomm) [inline test/ajv.test.js:147]:
```suggestion
  t.equal(response.json().message, 'response should be an object with an integer property answer only')
```

File: test/ajv.test.js [hunk 3]
```diff
@@ -74,3 +77,78 @@ test('should throw an error if ajv.plugins is not an array', async t => {
...
+          properties: {
+            answer: { type: 'number' }
+          },
+          additionalProperties: false,
+          errorMessage: 'should be an object with an integer property answer only'
+        }
+      }
+    },
+    handler: async (req, reply) => {
+      return { foo: 24 }
```
Comment (Eomm) [inline test/ajv.test.js:137]:
```suggestion
      return { notAnAnswer: 24 }
```

**Review by Eomm — COMMENTED** [review]
Minor suggestion before merging 🚀

File: test/ajv.test.js (hunk 2 above)
Comment (dancastillo) [reply test/ajv.test.js]: Should be updated now

File: test/ajv.test.js (hunk 3 above)
Comment (dancastillo) [reply test/ajv.test.js]: Should be updated now

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Can you resolve the conflicts?

**Comment by dancastillo** [comment]
@mcollina should be all set now

**Review by Eomm — APPROVED** [review]

**Comment by felixfbecker** [comment]
This is awesome, but unfortunately didn't update the TypeScript types

**Comment by felixfbecker** [comment]
https://github.com/fastify/fastify-response-validation/pull/96

**Pull request merged**